### PR TITLE
Improves unit tests coverage regarding Hibernate ORM's handling of import.sql imports #558 

### PR DESCRIPTION
**Pull request by machi1990**
Four test cases are covered:

- No load script is specified (defaults to `import.sql`)
-`import.sql` is specified as a load script file and it is present 
- a different sql file is specified as a load script and is present
- a different sql file is specified as a load script but it is absent

@Sanne @emmanuelbernard @gsmet

### CODE REVIEW

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java [hunk 1]
```diff
@@ -404,7 +404,6 @@ private void handleHibernateORMWithNoPersistenceXml(
                 // sql-load-script
                 // explicit file or default one
                 String file = hibernateConfig.sqlLoadScript.orElse("import.sql"); //default Hibernate ORM file imported
-
```
Comment (Sanne) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java]: could you remove this change? Not every important but it seems unrelated with the PR :)

File: extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/DefaultSqlLoadScriptTestCase.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+public class DefaultSqlLoadScriptTestCase {
+    @RegisterExtension
+    static QuarkusUnitTest runner = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource("application.properties"));
+
+    @Test
+    public void testDefaultSqlLoadScriptTest() {
+    }
```
Comment (Sanne) [inline extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/DefaultSqlLoadScriptTestCase.java]: I think this should test something so to make sure that that specific configuration resource was applied, and we also need to make sure that it all actually booted successfully?

File: extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/ImportSqlLoadScriptTestCase.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+public class ImportSqlLoadScriptTestCase {
+    @RegisterExtension
+    static QuarkusUnitTest runner = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource("application-import-load-script-test.properties", "application.properties")
+                    .addAsResource("import.sql"));
+
+    @Test
+    public void testImportSqlLoadScriptTest() {
+    }
```
Comment (Sanne) [inline extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/ImportSqlLoadScriptTestCase.java]: some here. How can we make sure that:

- it actually started Hibernate ORM successfully
- it didn't use the default configuration
- the specified `import.sql` script was run

**Review by Sanne — CHANGES_REQUESTED** [review]
It's a great start, thanks!

Added some suggestions, I think we need to make sure those tests are being run as intended - especially to make them more robust towards future regressions.

File: extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/DefaultSqlLoadScriptTestCase.java (hunk 2 above)
Comment (machi1990) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/DefaultSqlLoadScriptTestCase.java]: Yes. I was thinking of persisting an entity using the load script and make sure that using a `Entity.findById()` returns something.   Will this be enough in this case?

File: extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/ImportSqlLoadScriptTestCase.java (hunk 3 above)
Comment (machi1990) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/ImportSqlLoadScriptTestCase.java]: - The fact that `QuarkusUnitTest` runner did not throw was enough to make sure that the load-script did not cause the crash. Probably we can add something here? 

For second and last points, this test need to be enriched with verification that there is something loaded in the database: I'll add this part later on this evening.

Comment (machi1990) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/ImportSqlLoadScriptTestCase.java]: The same verifications can be added when loading a different sql script: `SqlLoadScriptPresentTestCase `.

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java (hunk 1 above)
Comment (machi1990) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java]: Good catch.  A mistake on my end. :-)

File: extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/DefaultSqlLoadScriptTestCase.java (hunk 2 above)
Comment (Sanne) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/DefaultSqlLoadScriptTestCase.java]: right I think that would be great.

Comment (machi1990) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/DefaultSqlLoadScriptTestCase.java]: Assertions are that the script is configuration is applied.

File: extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/ImportSqlLoadScriptTestCase.java (hunk 3 above)
Comment (machi1990) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/sql_load_scirpt/ImportSqlLoadScriptTestCase.java]: Done.

**Comment by stuartwdouglas** [comment]
@Sanne is this ok to merge now?

**Review by Sanne — APPROVED** [review]

**Comment by Sanne** [comment]
just reviewed again. Thanks @machi1990 , looks great! Merging..

**Comment by Sanne** [comment]
resolves #558
